eras/dijkstra/testlib/Test/Cardano/Ledger/Dijkstra/CDDL.hs

```diff
@@ -0,0 +1,476 @@
+      , opt $ idx 1 ==> nonempty_set native_script
+      , opt $ idx 2 ==> nonempty_set bootstrap_witness
+      , opt $ idx 3 ==> nonempty_set plutus_v1_script
+      , opt $ idx 4 ==> nonempty_set plutus_data
+      , opt $ idx 5 ==> redeemers
+      , opt $ idx 6 ==> nonempty_set plutus_v2_script
+      , opt $ idx 7 ==> nonempty_set plutus_v3_script
+      ]
+
+redeemers :: Rule
+redeemers =
+  comment
+    [str|Flat Array support is included for backwards compatibility and
+        |will be removed in the next era. It is recommended for tools to
+        |adopt using a Map instead of Array going forward.
+        |]
+    $ "redeemers"
+      =:= sarr
+        [ 1
+            <+ a
+              ( arr
+                  [ "tag" ==> redeemer_tag
+                  , "index" ==> (VUInt `sized` (4 :: Word64))
+                  , "data" ==> plutus_data
+                  , "ex_units" ==> ex_units
+                  ]
+              )
+        ]
+      / smp
+        [ 1
+            <+ asKey
+              ( arr
+                  [ "tag" ==> redeemer_tag
+                  , "index" ==> (VUInt `sized` (4 :: Word64))
+                  ]
+              )
+            ==> arr ["data" ==> plutus_data, "ex_units" ==> ex_units]
+        ]
+
+-- TODO: adjust with new script purpose
+redeemer_tag :: Rule
+redeemer_tag =
+  "redeemer_tag"
+    =:= (int 0 //- "spend")
+    / (int 1 //- "mint")
+    / (int 2 //- "cert")
+    / (int 3 //- "reward")
+    / (int 4 //- "voting")
+    / (int 5 //- "proposing")
+
+-- TODO: add Plutus V4
+language :: Rule
+language =
+  "language"
+    =:= int 0 -- Plutus v1
+    / int 1 -- Plutus v2
+    / int 2 -- Plutus v3
+
+-- TODO: add entry for Plutus v4
+cost_models :: Rule
+cost_models =
+  comment
+    [str|The format for cost_models is flexible enough to allow adding
+        |Plutus built-ins and language versions in the future.
+        |
+        |Plutus v1: only 166 integers are used, but more are accepted (and ignored)
+        |Plutus v2: only 175 integers are used, but more are accepted (and ignored)
+        |Plutus v3: only 223 integers are used, but more are accepted (and ignored)
+        |
+        |Any 8-bit unsigned number can be used as a key.
+        |]
+    $ "cost_models"
+      =:= mp
+        [ opt $ idx 0 ==> arr [0 <+ a int64]
+        , opt $ idx 1 ==> arr [0 <+ a int64]
+        , opt $ idx 2 ==> arr [0 <+ a int64]
+        , 0 <+ asKey ((3 :: Integer) ... (255 :: Integer)) ==> arr [0 <+ a int64]
+        ]
+
+-- TODO: add entry for Plutus v4
+alonzo_auxiliary_data :: Rule
+alonzo_auxiliary_data =
+  "alonzo_auxiliary_data"
+    =:= tag
+      259
+      ( mp
+          [ opt (idx 0 ==> metadata)
+          , opt (idx 1 ==> arr [0 <+ a native_script])
+          , opt (idx 2 ==> arr [0 <+ a plutus_v1_script])
+          , opt (idx 3 ==> arr [0 <+ a plutus_v2_script])
+          , opt (idx 4 ==> arr [0 <+ a plutus_v3_script])
+          ]
+      )
```

Comment:
This also similarly looks like something to be imported from the previous era. If we are adding a new thing to it, then it is probably confusing to name it on the previous eras I think, it becomes completely new?

Reply:
Same reason as above, it's `AlonzoTxAuxData` but it references the script type family, which will be different. As I said before, I can import it - just thought it will bring value to redefine the parts that will change.